### PR TITLE
Finalization committee parameters

### PR DESCRIPTION
**Pull request by EmilLa1**
## Purpose

Addresses  https://github.com/Concordium/concordium-node/issues/740 in the node.

## Changes

Added `FinalizationCommitteeParameters` to the set of update queues and handing of it in the `Scheduler.` 

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs [hunk 1]
```diff
@@ -1129,6 +1141,116 @@ processTimeParametersUpdates t bu = do
...
+                    let newConsensusParameters = case oldCP ^. cpConsensusParameters of
+                            cp@ConsensusParametersV1{} -> cp & cpBlockEnergyLimit .~ newBlockEnergyLimit
+                    newParameters <- refMake $ StoreSerialized $ oldCP & (cpConsensusParameters .~ newConsensusParameters)
+                    refMake
+                        u
+                            { currentParameters = newParameters,
+                              pendingUpdates = pendingUpdates{pBlockEnergyLimitQueue = SomeParam newpQ}
+                            }
+
+-- |Process min block time updates.
```
Comment (EmilLa1) [inline concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs:1225]: comment is off

**Review by td202 — APPROVED** [review]

File: concordium-consensus/src/Concordium/GlobalState/DummyData.hs [hunk 2]
```diff
@@ -359,6 +359,15 @@ dummyConsensusParametersV1 =
           _cpBlockEnergyLimit = maxBound -- the maximum energy a block can consume.
         }
 
+dummyFinalizationCommitteeParameters :: FinalizationCommitteeParameters
```
Comment (Bargsteen) [inline concordium-consensus/src/Concordium/GlobalState/DummyData.hs:362]: Please describe the concrete values used in a doc comment.

File: concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs [hunk 3]
```diff
@@ -1106,6 +1141,116 @@ processTimeParametersUpdates t bu = do
                               pendingUpdates = pendingUpdates{pTimeParametersQueue = SomeParam newpQ}
                             }
 
+-- |Process timeout parameters updates.
+processTimeoutParametersUpdates ::
```
Comment (Bargsteen) [inline concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs:1145]: Please add a few more details about what it does. Or refer to another similar function which has better documentation.

Comment (Bargsteen) [reply concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs:1145]: Same goes for the ones below.

**Review by Bargsteen — COMMENTED** [review]
Looks good in general, but I would like a bit more documentation for the non-trivial things 😊 (They might be trivial if you are very familiar with the existing setup, but it would be good to at least refer to more info).

**Review by abizjak — APPROVED** [review]
Looks fine, assuming you address Kasper's comments :)

File: concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs (hunk 3 above)
Comment (EmilLa1) [reply concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs:1145]: Added documentation for the ones requested, but I won't be starting documenting the whole module now.

**Pull request merged**